analytical_engine/core/grape_instance.cc

```diff
@@ -728,11 +688,114 @@ bl::result<std::string> GrapeInstance::contextToVineyardDataFrame(
   return toJson({{"object_id", s_id}});
 }
 
+bl::result<void> GrapeInstance::outputContext(const rpc::GSParams& params) {
```

Comment:
Overall LGTM, you can extract method to fix the CodeFactor issue and make the code more readable.